client/blocks/upsell-nudge/index.jsx

```diff
@@ -56,11 +56,14 @@ export const UpsellNudge = ( {
 		return null;
 	}
 
-	if ( ! href && site ) {
-		href = addQueryArgs( { feature, plan }, `/plans/${ site.slug }` );
-	}
+	const customerType = plan === PLAN_BUSINESS || PLAN_ECOMMERCE ? 'business' : 'personal';
+
+	const link =
+		href && site
+			? href
+			: addQueryArgs( { feature, customerType }, `/plans/${ site.slug }?${ plan }` );
```

Comment:
Why don't we add the plan with `addQueryArgs` as well?
```suggestion
			: addQueryArgs( { feature, customerType, plan }, `/plans/${ site.slug }` );
```